src/System.IdentityModel.Tokens.Jwt/JwtConstants.cs

```diff
@@ -47,9 +47,9 @@ public static class JwtConstants
         public const string TokenTypeAlt = "urn:ietf:params:oauth:token-type:jwt";
 
         /// <summary>
-        /// Token format: 'header.payload.signature'. Signature is optional, but '.' is required.
+        /// Token format: 'header.payload.signature'. Signature is optional, and if missing, trailing '.' is also optional.
         /// </summary>
-        public const string JsonCompactSerializationRegex = @"^[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\.[A-Za-z0-9-_]*$";
+        public const string JsonCompactSerializationRegex = @"^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)\.(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)\.?(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$";
```

Comment:
@brentschmaltz Here are a couple examples:

```
token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJTSEEyNTZ3aXRoUlNBIiwieDV0IjoiWldVNVlqazRNVEV3WmpsaVptRm1ZMk0yT0RsbVpUWmlPVEptWWpaa01ESTJOell6WWpBeU1BPT0ifQ==.eyJpc3MiOiJ3c28yLm9yZy9wcm9kdWN0cy9hbSIsImV4cCI6MTQzNDQ4NjQyNjM5NywiaHR0cDovL3dzbzIub3JnL2NsYWltcy9zdWJzY3JpYmVyIjoieGJibHRuYyIsImh0dHA6Ly93c28yLm9yZy9jbGFpbXMvYXBwbGljYXRpb25pZCI6Ijg0MSIsImh0dHA6Ly93c28yLm9yZy9jbGFpbXMvYXBwbGljYXRpb25uYW1lIjoiRGVmYXVsdEFwcGxpY2F0aW9uIiwiaHR0cDovL3dzbzIub3JnL2NsYWltcy9hcHBsaWNhdGlvbnRpZXIiOiJVbmxpbWl0ZWQiLCJodHRwOi8vd3NvMi5vcmcvY2xhaW1zL2FwaWNvbnRleHQiOiIvaW1hZ2VuZXciLCJodHRwOi8vd3NvMi5vcmcvY2xhaW1zL3ZlcnNpb24iOiJ2MSIsImh0dHA6Ly93c28yLm9yZy9jbGFpbXMvdGllciI6IkJyb256ZSIsImh0dHA6Ly93c28yLm9yZy9jbGFpbXMva2V5dHlwZSI6IlNBTkRCT1giLCJodHRwOi8vd3NvMi5vcmcvY2xhaW1zL3VzZXJ0eXBlIjoiQVBQTElDQVRJT04iLCJodHRwOi8vd3NvMi5vcmcvY2xhaW1zL2VuZHVzZXIiOiJ4YmJsdG5jIiwiaWF0IjoxNDM0NDg1NTI2Mzk3fQ==.AgWYgHeMqcSQQQIhflvZlsa2LLKIKjhAD/2lDH82SB/27FoodZk4CSfzXnQh9DDZGXWHfxuCDH18TlRY6WpH7HCOKZRre3dbnxx0+hbjX8iUUk1XggQIGJTRXBHpm35ZqQcDJHxJ/3Kw+xvGJT0sopukbjbMhdZVQzSkfLNpDxA=";
token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJTSEEyNTZ3aXRoUlNBIiwieDV0IjoiWldVNVlqazRNVEV3WmpsaVptRm1ZMk0yT0RsbVpUWmlPVEptWWpaa01ESTJOell6WWpBeU1BPT0ifQ==.eyJpc3MiOiJ3c28yLm9yZy9wcm9kdWN0cy9hbSIsImV4cCI6MTQ0NDgwNjg3Mjk3NCwiaHR0cDovL3dzbzIub3JnL2NsYWltcy9zdWJzY3JpYmVyIjoieGJiajlmeSIsImh0dHA6Ly93c28yLm9yZy9jbGFpbXMvYXBwbGljYXRpb25pZCI6IjE1NzgiLCJodHRwOi8vd3NvMi5vcmcvY2xhaW1zL2FwcGxpY2F0aW9ubmFtZSI6IlNlY3VyaXRpZXNEYXRhU2VydmljZUFwcGxpY2F0aW9uIiwiaHR0cDovL3dzbzIub3JnL2NsYWltcy9hcHBsaWNhdGlvbnRpZXIiOiJVbmxpbWl0ZWQiLCJodHRwOi8vd3NvMi5vcmcvY2xhaW1zL2FwaWNvbnRleHQiOiIvc2RzIiwiaHR0cDovL3dzbzIub3JnL2NsYWltcy92ZXJzaW9uIjoidjEiLCJodHRwOi8vd3NvMi5vcmcvY2xhaW1zL3RpZXIiOiJVbmxpbWl0ZWQiLCJodHRwOi8vd3NvMi5vcmcvY2xhaW1zL2tleXR5cGUiOiJQUk9EVUNUSU9OIiwiaHR0cDovL3dzbzIub3JnL2NsYWltcy91c2VydHlwZSI6IkFQUExJQ0FUSU9OIiwiaHR0cDovL3dzbzIub3JnL2NsYWltcy9lbmR1c2VyIjoieGJiajlmeSIsImlhdCI6MTQ0NDgwNTk3Mjk3NH0=.MMfZeYcfOLYwaMBP1M7oY4WpplROt7L5t+gtzAtQzB4ktYBCMest7TsGYvgOzN8agjGOijYlIG33N8HgEkP4KPi3Ajpcgxp6rD4jeKIIeU3NeeGGYVHPOCCww3R8uHD4mJI8QgZRO0cDoeq4ZWpzS///e4v/GUSY+CN8J6om6VY=";
```
The above regex was created from base64 regex referenced here:
http://stackoverflow.com/questions/475074/regex-to-parse-or-validate-base64-data